modules/tftp/server.rb

```diff
@@ -150,6 +150,26 @@ def pxeconfig_file(mac)
     end
   end
 
+  def self.fetch_boot_image(image_dst, url, files)
+    # Verify dst is a valid directory
+    image_path = Pathname.new(image_dst).cleanpath
+    extr_image_dir = Pathname.new(image_dst.delete_suffix(".iso"))
+
+    FileUtils.mkdir_p image_path.parent
+    choose_protocol_and_fetch(url, image_path).join
+
+    files.each do |file|
+      file_path = Pathname.new file
+      extr_file_path = Pathname.new(File.join(extr_image_dir, file_path)).cleanpath
+
+      # Create destination directory
+      FileUtils.mkdir_p extr_file_path.parent
+      # extract iso
+      extract_task = ::Proxy::ArchiveExtract.new(image_path, file_path, extr_file_path).start
+      raise "TFTP image file extraction error: #{file_path}" unless extract_task.join == 0
```

Comment:
This does mean that if a single file couldn't be extracted, you leave somewhere in the middle. Not sure how you would do this well though. Making this transactional is hard.

Reply:
Do you think it would be better to continue with extracting the other files? We could check in the end whether all files have been extracted and return an error message depending on that.